### ISSUE TITLE
Following upgrade from pg 4.3.0 to pg 7.11.0 unable to authenticate with PostgreSQL docker container

### ISSUE THREAD

**Issue by MelissaSnell** [brianc/node-postgres#1899]
Hi there
Our code base was running using pg 4.3.0 and was able to access a PostgreSQL database in a Docker container without issue.

Since updating the pg module to 7.11.0 and setting up the initial connection using the pg.pool, any code that contacts the database e.g. a pg.query fails with an authentication error.

Code is running on a Mac.

The Docker container for local development is created by running manually :
docker run -p 5432:5432 --name container-postgres -e POSTGRES_DB=test_postgres_db -e POSTGRES_PASSWORD=test_user -e POSTGRES_USER=test_password -d postgres

and then starting the container.

The Docker container is running PostgreSQL 11.2.

The code is running node version 8.5.

From my local development logs I can see the message:
`Error executing query error: password authentication failed for user "xxxxxxxx"`

The user name displayed is the one in the Mac "USER" environment variable, so presumably the root user for the PC?

The code connects to the pool like this (environment variables are set prior to reaching this point):

`let connectionInfo = config.PostgreSQL.newConnectionInfo ? config.PostgreSQL.newConnectionInfo : null;
if (!connectionInfo){
    connectionInfo = {
        host : process.env.POSTGRES_HOST,
        port : process.env.POSTGRES_PORT,
        user : process.env.POSTGRES_USER,
        password : process.env.POSTGRES_PASSWORD,
        database : process.env.POSTGRES_DATABASE,
        ssl : {},
        max : 20, // max number of clients in the pool
    };
}

logger.info(`Creating database connection pool using ${JSON.stringify(connectionInfo)}`);

const pool = new Pool({ connectionInfo });
logger.info('Connected to PostgreSQL database');`

If I stop Docker and instead just launch a local PostgreSQL server, the code runs without issue, so it must be something to do with how the code is now communicating with Docker.

The reason that I am raising it as an issue here is that prior to upgrading the pg version, it worked without issue.

Is there an additional setting/ step I need to employ to get this to work in Docker? It needs to , because I am using Codeship for deployments.

In Codeship build failure log I am seeing this error there:

Error executing query Error: connect ECONNREFUSED 127.0.0.1:5432

Again, the Codeship build used to work without issue prior to the upgrade.

Any suggestions would be gratefully received. 

Many thanks in advance.

**Comment by MelissaSnell** [comment]
I have just noted that in the documentation that:

PGHOST='localhost'
PGUSER=process.env.USER
PGDATABASE=process.env.USER
PGPASSWORD=null
PGPORT=5432

https://node-postgres.com/features/connecting

I am going to go back and double-check the connection information that I am passing in when I create the pool, because it looks like this is being ignored.

**Comment by MelissaSnell** [comment]
More information.

If I run "env" on the mac then I can't see the environment variables set by the code, but I added logging statement to the code to display the value of :
POSTGRES_USER
POSTGRES_PASSWORD

at runtime, and they were displaying the correct values.

If instead I manually export 
PGUSER
PGPASSWORD
PGDATABASE
prior to a test run then the code can access the Docker container PostgreSQL without error.

So the issue is either:
a] I thought that I could run pool.connect once at the beginning. The pool creation and connect is located in a separate lib/db.js file which is then "required" in all the modules where I need to connect to the database. Again, as this seemed to work fine with a local PostgreSQL database I thought this was OK
b] there is an issue with Pool where it is not actually using the connection details that are being passed in to make the connection - instead it is automatically using the defaulf.
c] I am doing something wrong in the connectionInfo object that I am passing in.
d] I will have to update our systems to use a new environment variable set to move to PGUSER, PGPASSWORD etc.

So in summary, why is the Pool connect not using the information being passed to it?

**Comment by charmander** [comment]
```js
const pool = new Pool({ connectionInfo });
```

should be

```js
const pool = new Pool(connectionInfo);
```

**Comment by MelissaSnell** [comment]
Many thanks for pointing out my error. I can confirm that everything is working as expected now. The thing that threw me was the fact that the local database access was working.
